tests/v3_api/test_upgrade.py

```diff
@@ -573,3 +591,52 @@ def upgrade_rancher_server(serverIp,
           sshUser, sshKeyPath))
 
     wait_until_active(CATTLE_TEST_URL)
+
+
+def create_and_validate_catalog_app():
+    cluster = namespace["cluster"]
+    p_client = namespace['p_client']
+    name = create_prefix + app_name
+    new_ns = create_ns(get_cluster_client_for_token(cluster, ADMIN_TOKEN),
+                       cluster, namespace["project"], ns_name=name)
+    app = p_client.create_app(
+        answers={
+            "defaultImage": "true",
+            "image": "mysql",
+            "imageTag": "5.7.14",
+            "mysqlDatabase": "admin",
+            "mysqlPassword": "",
+            "mysqlUser": "admin",
+            "persistence.enabled": "false",
+            "persistence.size": "8Gi",
+            "persistence.storageClass": "",
+            "service.nodePort": "",
+            "service.port": "3306",
+            "service.type": "ClusterIP"
+        },
+        externalId="catalog://?catalog=library&template=mysql&version=0.3.7",
+        name=create_prefix + app_name,
+        projectId=namespace["project"].id,
+        prune=False,
+        targetNamespace=new_ns.id
+    )
+    validate_catalog_app(app.name)
+
+
+def modify_catalog_app(app_name):
+    p_client = namespace["p_client"]
+    app = wait_for_app_to_active(p_client, app_name)
+    externalId = "catalog://?catalog=library&template=mysql&version=0.3.8"
+    p_client.action(obj=app, action_name="upgrade", externalId=externalId)
+    validate_catalog_app(app_name)
```

Comment:
@jiaqiluo Could you add some more checks (may be image name if that changes between 2 versions) on the upgraded app other than status being "active" ?

Reply:
done